### ISSUE TITLE
[Request] Scripting-API - Allow for tool-groups / sections

### ISSUE THREAD

**Issue by BlaXun** [mapeditor/tiled#3242]
I think it would be nice if tools could be grouped in a section. The section could have a name that would be shown in the toolbar above/below the actual tools. This way plugins could easily be distinguished from each other

**Comment by bjorn** [comment]
I guess you mean similar to how it is [done here](https://stackoverflow.com/questions/26837441/how-to-give-headings-for-each-toolbar-section-in-qt)?

![image](https://user-images.githubusercontent.com/531764/149916529-fdfb8eb3-43a1-4652-a253-4969eba424bf.png)

It's a nice idea, but I think it would easily look a bit cluttered. As a first step, it might be a good idea to add support for adding separators between tools.

**Comment by BlaXun** [comment]
Yeah, like that. Separators would be a good starting point I assume. Maybe
upon calling registerTool a group identifier could be given to add the tool
to the specific group.
Ideally the user would have a way to disable specific groups in order to
not have a cramped toolbar.

**Comment by eishiya** [comment]
I'm not interested in labels, but I'd love defined sections that let you insert a tool at a specific location (similar logic to e.g. `map.addLayerAt(index)`). Tools are added to the toolbar in the order their script files are read, which means if you have multiple tools, they'll likely end up not organized in any meaningful way. With menus, at least you can control where in the menu the item will go, though there are still load order issues if you want to group items from multiple scripts together. With tools, there's no control at all.
